app/models/compute_resource.rb

```diff
@@ -169,8 +169,8 @@ def new_interface(attr = {})
   end
 
   # return a list of virtual machines
-  def vms(opts = {})
-    client.servers
+  def vms(attrs = {})
+    client.servers(attrs)
```

Comment:
How confident we are that `client.servers` accept arguments for each compute resource? I see the client is always defined by the fog adapter. Did you go through all adapters we use?

Otherwise looks good

Reply:
I did not go through all adapters, though I am pretty confident, because all of them should define servers as a `collection` what gives them the attrs argument.
See the definition of the collection method: https://github.com/fog/fog-core/blob/a60d4001981b8d4ca9a803ae9cac07f3a99fad23/lib/fog/core/service.rb#L282

Reply:
Though I do not know if there is not some adapter defining the method in some shady way, I am pretty confident there should not be.

Reply:
makes sense, thanks!